lib/ansible/modules/network/meraki/meraki_static_route.py

```diff
@@ -0,0 +1,386 @@
+          description: First address in reservation range, inclusive.
+          returned: query or update
+          type: string
+          sample: 192.0.1.2
+        end:
+          description: Last address in reservation range, inclusive.
+          returned: query or update
+          type: string
+          sample: 192.0.1.10
+        comment:
+          description: Human readable description of range.
+          returned: query or update
+          type: string
+          sample: Server range
+    fixedIpAssignments:
+      description: List of static MAC to IP address bindings.
+      returned: query or update
+      type: complex
+      contains:
+        mac:
+          description: Key is MAC address of endpoint.
+          returned: query or update
+          type: complex
+          contains:
+            ip:
+              description: IP address to be bound to the endpoint.
+              returned: query or update
+              type: string
+              sample: 192.0.1.11
+            name:
+              description: Hostname given to the endpoint.
+              returned: query or update
+              type: string
+              sample: JimLaptop
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule, json, env_fallback
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_native
+from ansible.module_utils.network.meraki.meraki import MerakiModule, meraki_argument_spec
+
+
+def fixed_ip_factory(meraki, data):
+    fixed_ips = dict()
+    for item in data:
+        fixed_ips[item['mac']] = {'ip': item['ip'], 'name': item['name']}
+    return fixed_ips
+
+
+def get_static_routes(meraki, net_id):
+    path = meraki.construct_path('get_all', net_id=net_id)
+    r = meraki.request(path, method='GET')
+    return r
+
+
+def get_static_route(meraki, net_id, route_id):
+    path = meraki.construct_path('get_one', net_id=net_id, custom={'route_id': meraki.params['route_id']})
+    r = meraki.request(path, method='GET')
+    return r
+
+
+def main():
+
+    # define the available arguments/parameters that a user can pass to
+    # the module
+
+    fixed_ip_arg_spec = dict(mac=dict(type='str'),
+                             ip=dict(type='str'),
+                             name=dict(type='str'),
+                             )
+
+    reserved_ip_arg_spec = dict(start=dict(type='str'),
+                                end=dict(type='str'),
+                                comment=dict(type='str'),
+                                )
+
+    argument_spec = meraki_argument_spec()
+    argument_spec.update(
+        net_id=dict(type='str'),
+        net_name=dict(type='str'),
+        name=dict(type='str'),
+        subnet=dict(type='str'),
+        gateway_ip=dict(type='str'),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        fixed_ip_assignments=dict(type='list', default=None, elements='dict', options=fixed_ip_arg_spec),
```

Comment:
Strictly speaking the `default=None` is not needed.

Reply:
Fixed